### PR TITLE
opentelemetry-collector: init at 0.26.0

### PR DESCRIPTION
**Pull request by uri-canva**
###### Motivation for this change

Add https://github.com/open-telemetry/opentelemetry-collector/.

Note this is built using the source at https://github.com/open-telemetry/opentelemetry-collector-contrib/ rather than the main project, as it has more receivers and exporters. It has the same license and is a strict superset as noted in the description:
> Please note that this repository and its releases are a superset of the core repository.

However I have left the binary name as `otelcontribcol`.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/opentelemetry-collector/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    sha256 = "03713b4bkhcz61maz0r5mkd36kv3rq8rji3qcpi9zf5bkkjs1yzb";
+  };
+
+  vendorSha256 = "0anw3l6pq8yys2g2607ndhklb9m1i9krgjrw4wb99igavjzp3wpj";
+
+  subPackages = [ "cmd/otelcontribcol" ];
+
+  meta = with lib; {
+    homepage = "https://github.com/open-telemetry/opentelemetry-collector";
+    description = " OpenTelemetry Collector";
```
Comment (legendofmiracles) [inline pkgs/tools/misc/opentelemetry-collector/default.nix:23]:
```suggestion
    description = "OpenTelemetry Collector";
```

File: pkgs/tools/misc/opentelemetry-collector/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    license = licenses.asl20;
+    maintainers = [ maintainers.uri-canva ];
+    platforms = platforms.darwin ++ platforms.linux;
```
Comment (legendofmiracles) [inline pkgs/tools/misc/opentelemetry-collector/default.nix:26]:
```suggestion
```
This is set by buildGoModule

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123232` at 43975052 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/17470b2a2e3625af954de70fffb27a64">opentelemetry-collector</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/tools/misc/opentelemetry-collector/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchFromGitHub {
+    owner = "open-telemetry";
+    repo = "${pname}-contrib";
+    rev = "v${version}";
+    sha256 = "03713b4bkhcz61maz0r5mkd36kv3rq8rji3qcpi9zf5bkkjs1yzb";
+  };
+
+  vendorSha256 = if stdenv.isDarwin
+    then "0anw3l6pq8yys2g2607ndhklb9m1i9krgjrw4wb99igavjzp3wpj"
+    else "04h463d2d7g6wqp5mzkqlszwzdbq0pix6j7n2s9s80lwg7nh8k3h";
```
Comment (uri-canva) [inline pkgs/tools/misc/opentelemetry-collector/default.nix:20]: Is it ok to have this? No other `buildGoModule` has this split as far as I can tell.

Comment (SuperSandro2000) [reply pkgs/tools/misc/opentelemetry-collector/default.nix:20]: It shouldn't be needed to be honest.

File: pkgs/tools/misc/opentelemetry-collector/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+, lib
+}:
+
+buildGoModule rec {
+  pname = "opentelemetry-collector";
+  version = "0.26.0";
+
+  src = fetchFromGitHub {
+    owner = "open-telemetry";
+    repo = "${pname}-contrib";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/opentelemetry-collector/default.nix:13]:
```suggestion
    repo = "opentelemetry-collector-contrib";
```

File: pkgs/tools/misc/opentelemetry-collector/default.nix (hunk 3 above)
Comment (uri-canva) [reply pkgs/tools/misc/opentelemetry-collector/default.nix:20]: They are different though, I'm not sure why. Maybe a lot of the other `buildGoModule` don't support darwin so it didn't come up before.

**Comment by lukegb** [comment]
@ofborg build opentelemetry-collector opentelemetry-collector.passthru.tests

**Comment by lukegb** [comment]
```
lukegb@Mac-mini pr123232 % ./result/bin/otelcontribcol --version
otelcontribcol version latest
```

**Pull request merged**